src/storage/compression/numeric_constant.cpp

```diff
@@ -0,0 +1,158 @@
+}
+
+//===--------------------------------------------------------------------===//
+// Fetch
+//===--------------------------------------------------------------------===//
+void ConstantFetchRowValidity(ColumnSegment &segment, ColumnFetchState &state, row_t row_id, Vector &result,
+                              idx_t result_idx) {
+	FillFunctionValidity(segment, result, result_idx, 1);
+}
+
+template <class T>
+void ConstantFetchRow(ColumnSegment &segment, ColumnFetchState &state, row_t row_id, Vector &result, idx_t result_idx) {
+	FillFunction<T>(segment, result, result_idx, 1);
+}
+
+//===--------------------------------------------------------------------===//
+// Get Function
+//===--------------------------------------------------------------------===//
+CompressionFunction ConstantGetFunctionValidity(PhysicalType data_type) {
+	D_ASSERT(data_type == PhysicalType::BIT);
+	return CompressionFunction(CompressionType::COMPRESSION_CONSTANT, data_type, nullptr, nullptr, nullptr, nullptr,
+	                           nullptr, nullptr, ConstantInitScan, ConstantScanFunctionValidity,
+	                           ConstantScanPartialValidity, ConstantFetchRowValidity, UncompressedFunctions::EmptySkip,
+	                           nullptr, nullptr, nullptr);
+}
+
+template <class T>
+CompressionFunction ConstantGetFunction(PhysicalType data_type) {
+	return CompressionFunction(CompressionType::COMPRESSION_CONSTANT, data_type, nullptr, nullptr, nullptr, nullptr,
+	                           nullptr, nullptr, ConstantInitScan, ConstantScanFunction<T>, ConstantScanPartial<T>,
+	                           ConstantFetchRow<T>, UncompressedFunctions::EmptySkip, nullptr, nullptr, nullptr);
+}
+
+CompressionFunction ConstantFun::GetFunction(PhysicalType data_type) {
+	switch (data_type) {
+	case PhysicalType::BIT:
+		return ConstantGetFunctionValidity(data_type);
+	case PhysicalType::BOOL:
+	case PhysicalType::INT8:
+		return ConstantGetFunction<int8_t>(data_type);
+	case PhysicalType::INT16:
+		return ConstantGetFunction<int16_t>(data_type);
+	case PhysicalType::INT32:
+		return ConstantGetFunction<int32_t>(data_type);
+	case PhysicalType::INT64:
+		return ConstantGetFunction<int64_t>(data_type);
+	case PhysicalType::UINT8:
+		return ConstantGetFunction<uint8_t>(data_type);
+	case PhysicalType::UINT16:
+		return ConstantGetFunction<uint16_t>(data_type);
+	case PhysicalType::UINT32:
+		return ConstantGetFunction<uint32_t>(data_type);
+	case PhysicalType::UINT64:
+		return ConstantGetFunction<uint64_t>(data_type);
+	case PhysicalType::INT128:
+		return ConstantGetFunction<hugeint_t>(data_type);
+	case PhysicalType::FLOAT:
+		return ConstantGetFunction<float>(data_type);
+	case PhysicalType::DOUBLE:
+		return ConstantGetFunction<double>(data_type);
+	default:
+		throw InternalException("Unsupported type for ConstantUncompressed::GetFunction");
+	}
+}
+
+bool ConstantFun::TypeIsSupported(PhysicalType type) {
```

Comment:
is this a numeric method or more generic?

Reply:
Numeric only (for now)